### ISSUE TITLE
Add DANS Endpoint

### ISSUE THREAD

**Issue by john-shepherdson** [cessda/cessda.cdc.versions#667]
>  Hi everyone,
> I would like to solve this so that our metadata can be included in the CDC.
> Asking my colleagues they told me that we have to sets (an EN and a Dutch one) and I believe these were previously used to get our metadata into the CDC.
> https://ssh.datastations.nl/oai?verb=ListRecords&metadataPrefix=oai_dc&set=CESSDA-EN
> https://ssh.datastations.nl/oai?verb=ListRecords&metadataPrefix=oai_dc&set=CESSDA-NL
> 
> Do I understand it correctly that having these sets is not sufficient?
> Our metadata actually does include a "language of metadata" attribute which is even now made mandatory, but it might not be included in all of the metadata exports Dataverse provides.
> 
> If these sets are insufficient, could you give us more details about what would be required and what is harvested from Dataverse? Since other SPs using Dataverse are included in the CDC I hope we could adjust our exports to comply with your requirements.
> 
> Many thanks.
> Ricarda

_Originally posted by @RicardaBraukmann in https://github.com/cessda/cessda.cdc.versions/issues/662#issuecomment-2180919866_

> We could use the sets as above, but would have to treat them as 2 different endpoints with different names and different default languages. Might be confusing for the users to see publishes called (for example) 'DANS-KNAW (English)' and 'DANS-KNAW (Dutch)' - also the names would not comply with the Publisher names CV (https://vocabularies.cessda.eu/vocabulary/CdcPublisherNames?lang=en)

_Originally posted by @john-shepherdson in https://github.com/cessda/cessda.cdc.versions/issues/662#issuecomment-2181069028_

> Thanks @john-shepherdson for looking into it. For us I would prefer to be included in some way so as soon as possible so if what you say is possible that would be great. Alternatively, you could also for now harvest the English records only as those will be most relevant for CDC users I believe and that set is also our bigger set from the two. 
> 
> Of course we want to be included full as soon as possible so it would be great if  we can discuss how that can be achieved. 
> 
> Can you specify what we need to do in order to be harvested through our regular endpoint?
> 
> We have language of metadata information in our metadata in a custom block so the information is available for most datasets. I am not sure how you harvest the Dataverse instances (i.e. what metadata schema do you use), and what adjustments we would need to make to comply with the requirements? 
> I am happy to connect you with our technical team as well as they know better how things are currently implemented.

_Originally posted by @RicardaBraukmann in https://github.com/cessda/cessda.cdc.versions/issues/662#issuecomment-2185835141_
See also #662

**Comment by john-shepherdson** [comment]
Ricarda Braukmann wrote:
"Thanks @john-shepherdson for looking into it. For us I would prefer to be included in some way so as soon as possible so if what you say is possible that would be great. Alternatively, you could also for now harvest the English records only as those will be most relevant for CDC users I believe and that set is also our bigger set from the two.

Of course we want to be included full as soon as possible so it would be great if we can discuss how that can be achieved.

Can you specify what we need to do in order to be harvested through our regular endpoint?

We have language of metadata information in our metadata in a custom block so the information is available for most datasets. I am not sure how you harvest the Dataverse instances (i.e. what metadata schema do you use), and what adjustments we would need to make to comply with the requirements?
I am happy to connect you with our technical team as well as they know better how things are currently implemented."

**Comment by john-shepherdson** [comment]
@KristinaS4 @MortenSikt Your thoughts please.

**Comment by matthew-morris-cessda** [comment]
Added quotes from @RicardaBraukmann

**Comment by KristinaS4** [comment]
@john-shepherdson 
Do you know the status on the language tags for their Dataverse endpoint? It seems as they are willing to adjust their exports to comply with CDC's requirements. This would of course be the optimal solution. Can we do more to support this?

I agree that it will be confusing for users if there are two publishers called DANS-KNAW and in that case I think we should only include the English records as suggested by Ricarda to limit it to one publisher.

**Comment by john-shepherdson** [comment]
I am not aware that the language tags have been added (but maybe Matthew could have a look at some recently harvested DANS XML files to confirm) in which case the short term fix is only to include the records from the English endpoint.

**Comment by LauraHuisintveld** [comment]
Dear all, At DANS we do not have the language tags added to our DDI export via OAI-PMH. There is a Dataverse setting for it, but we have not enabled it. I will discuss with Ricarda what is the best way to solve the issue.

**Comment by alen-vodopijevec-cessda** [comment]
Dear @LauraHuisintveld @RicardaBraukmann

Following up on the e-mail exchange.. CESSDA Metadata Validator is available at: https://cmv.cessda.eu/#!validation

Please check it out for validating your outputs and get back to us when successful or if you have any issues in the meanwhile.

**Comment by LauraHuisintveld** [comment]
@alen-vodopijevec-cessda Yes, we will test our output with the validator, a very useful tool. One question, should we test with DDI Profile 2.0, or with 1.04 which is also still available within the validator?

**Comment by alen-vodopijevec-cessda** [comment]
You should use 1.04, passing this validation will guarantee the compliance with the CDC. You can also give it a try and test with 2.0 profile - just curious about the results as this is more strict.

**Comment by LauraHuisintveld** [comment]
Thanks, we can let you know our results once we are ready. 
I already played around with the validator a bit, and I have another question. Sometimes our users have used html-code within the <abstract>. In one case, this resulted in a schema violation. A <b> and <ul> element were used. 
Is this specific for the validator tool, or will the Data Catalogue not accept these records as well?

**Comment by matthew-morris-cessda** [comment]
The Data Catalogue will accept these records

**Comment by LauraHuisintveld** [comment]
Dear all,
We have tested our new OAI-PMH output, and we think it is ready now. Could you please test this link and let us know if works without problems?
https://oai-service.labs.dans.knaw.nl/ss/oai?verb=ListRecords&set=social_sciences&metadataPrefix=oai_ddi

**Comment by MortenSikt** [comment]
I can see in staging that 6749 records are found from DANS-KNAW. Some metadata is presented in the results list, but clicking on any record does not lead to a valid page (its just blank for me). 

Not sure if this is before or after configuration @matthew-morris-cessda ?

**Comment by matthew-morris-cessda** [comment]
I've only just implemented this configuration change so this isn't available yet. I'll update the issue when it is.

**Comment by LauraHuisintveld** [comment]
@matthew-morris-cessda I was wondering if there is any news? Are there still some problems we need to solve at our side?

**Comment by matthew-morris-cessda** [comment]
@LauraHuisintveld Dutch language content is still tagged as English on this endpoint

**Comment by LauraHuisintveld** [comment]
Hi @matthew-morris-cessda We have found the problem and did a new deploy. Could you please try again? The URL remains the same: https://oai-service.labs.dans.knaw.nl/ss/oai?verb=ListRecords&set=social_sciences&metadataPrefix=oai_ddi

**Comment by matthew-morris-cessda** [comment]
The problem still persists

![image](https://github.com/user-attachments/assets/d4ce243d-2961-43a9-99dc-4a6071416600)

**Comment by LauraHuisintveld** [comment]
@matthew-morris-cessda Yes, this is an example of a dataset that has no information about the metadata language. Therefore, it is automatically tagged as English. 
However, there should be around 400 datasets in Dutch metadata language, can you see them now? If so, then we know our code is working.

**Comment by matthew-morris-cessda** [comment]
@LauraHuisintveld Shouldn't it be better to either 1. Not tag metadata with no language information, or 2. Remove metadata with no language information from the export, rather than tagging wrong language information

**Comment by matthew-morris-cessda** [comment]
@LauraHuisintveld I can confirm that there were 443 datasets marked as Dutch

**Comment by RicardaBraukmann** [comment]
@matthew-morris-cessda if you can see these 443 datasets that should mean that our tool is working as expected.
Can you use this to include our data in the CDC?

We know that not all language classifications are correct (in particular some datasets are labelled English while being in Dutch) but this should be a minority. We have plans to improve this early next year.

But for now having the Data Station harvested rather than having the old entries in the CDC would be much preferred from our side.
Please let me know if it is possible to harvest.

**Comment by MortenSikt** [comment]
If it is just a minority that are wrongly tagged now and with DANS wishing to have the Data Station harvested (it perhaps has newer or more correct metadata @RicardaBraukmann ?).  I think we should switch from the old to the new, even though there might be some errors.

**Comment by RicardaBraukmann** [comment]
Yes it is for sure newer and more correct - the current CDC for DANS still links to EASY which we have replaced with our Data Stations and I would hope that with the new import we could at least ensure that users are directed to the DANS Data Station. Also as we mapped information to the CDC fields I believe the information should be more complete. I cannot say for sure how many records will be falsely tagged as English when in fact being in Dutch but generally most of our metadata is provided in English. And as said we plan to fix this in the near future.

**Comment by MortenSikt** [comment]
Thank you for clarifying. Can you switch to the new endpoint @matthew-morris-cessda ?

**Comment by MortenSikt** [comment]
Can we close this issue @matthew-morris-cessda ? Or should we keep it open until all records are correct?

**Comment by matthew-morris-cessda** [comment]
I think that this issue should remain open until all language issues are fixed.

**Comment by MortenSikt** [comment]
Not relevant for codebase upgrade, will be monitored outside of release schedule

**Comment by RicardaBraukmann** [comment]
Just to check, do I understand it correctly that you are able to harvest the metadata correctly and that it is ok for the CDC?

When would this be updated in the CDC? Currently the page the CDC refers to for the DANS datasets is still EASY (Access Study links to http://easy.dans.knaw.nl/). That is our old system and I would very much like this to be changed as soon as possible.

**Comment by matthew-morris-cessda** [comment]
Yes, we can harvest the metadata correctly. We're hoping to get a release of CDC with the new endpoint deployed in approx. 1 month for now.

**Comment by RicardaBraukmann** [comment]
Great, thanks for confirming @matthew-morris-cessda .

**Comment by MortenSikt** [comment]
see also: https://github.com/cessda/cessda.cdc.versions/issues/704 this causes issues for other records in the catalogue

**Comment by RicardaBraukmann** [comment]
@MortenSikt Can you elaborate what is needed from our side? 
Are changes required in what we provide to the CDC or can we leave things as they are with the future dates?


As @LauraHuisintveld pointed out in the other issue, it seems that in DDI the distDate can be in docDscr (publication date) and in StdyDscr (distribution date). The latter will be in the future if the embargo functionality is used. 
The embargo functionality is standard in Dataverse so it seems to me that anyone using the embargo functionality would have the same issue.

**Comment by RicardaBraukmann** [comment]
Thank you for clarifying. The language issue is on our list and we how to have the right language included for all the metadata soon, but it takes time for us as the field was not required in the past. 
And great if you can keep us updated on the CMM guidance and CDC profile and any changes with respect to the dates we should provide.
